test/lib/tasks/message_queue_test.rb

```diff
@@ -0,0 +1,22 @@
+require "test_helper"
+
+class MessageQueueConsumerRakeTest < ActiveSupport::TestCase
+  context "when consuming messages from RabbitMQ" do
+    should "use GovukMessageQueueConsumer::Consumer" do
+      tagging_updater = mock('TaggingUpdater')
+      TaggingUpdater.expects(:new).returns(tagging_updater)
+
+      consumer = mock('consumer')
+      consumer.expects(:run).returns(true)
+
+      GovukMessageQueueConsumer::Consumer.expects(:new)
```

Comment:
I'm not sure if I follow

If you changed the consumer API this test will fail

With a new argument (boom: "hello")
```
  1) Failure:
MessageQueueConsumerRakeTest#test_: when consuming messages from RabbitMQ should use GovukMessageQueueConsumer::Consumer.  [/var/govuk/panopticon/lib/tasks/message_queue.rake:4]:
unexpected invocation: GovukMessageQueueConsumer::Consumer.new(:queue_name => 'panopticon', :processor => #<Mock:TaggingUpdater>, :boom => 'hello')
unsatisfied expectations:
- expected exactly once, not yet invoked: #<Mock:consumer>.run(any_parameters)
- expected exactly once, not yet invoked: GovukMessageQueueConsumer::Consumer.new(:queue_name => 'panopticon', :processor => #<Mock:TaggingUpdater>)
satisfied expectations:
- expected exactly once, invoked once: TaggingUpdater.new(any_parameters)
```

If you change the queue_name

```
  1) Failure:
MessageQueueConsumerRakeTest#test_: when consuming messages from RabbitMQ should use GovukMessageQueueConsumer::Consumer.  [/var/govuk/panopticon/lib/tasks/message_queue.rake:4]:
unexpected invocation: GovukMessageQueueConsumer::Consumer.new(:queue_name => 'panopticon', :processor => #<Mock:TaggingUpdater>)
unsatisfied expectations:
- expected exactly once, not yet invoked: #<Mock:consumer>.run(any_parameters)
Press ENTER or type command to continue
- expected exactly once, not yet invoked: GovukMessageQueueConsumer::Consumer.new(:queue_name => 'changed', :processor => #<Mock:TaggingUpdater>, :boom => 'changed')
satisfied expectations:
- expected exactly once, invoked once: TaggingUpdater.new(any_parameters)
```

And this is what I intended to test, in case something changes I want this test to break.